contracts/src/v0.8/functions/dev/1_0_0/FunctionsBilling.sol

```diff
@@ -0,0 +1,446 @@
+      uint32 gasOverhead,
+      uint32 requestTimeoutSeconds,
+      uint96 donFee
+    ) = abi.decode(config, (uint32, uint32, uint256, int256, uint32, uint32, uint96));
+
+    if (fallbackNativePerUnitLink <= 0) {
+      revert InvalidLinkWeiPrice(fallbackNativePerUnitLink);
+    }
+    s_config = Config({
+      maxGasLimit: maxGasLimit,
+      stalenessSeconds: stalenessSeconds,
+      gasAfterPaymentCalculation: gasAfterPaymentCalculation,
+      gasOverhead: gasOverhead,
+      requestTimeoutSeconds: requestTimeoutSeconds,
+      donFee: donFee,
+      fallbackNativePerUnitLink: fallbackNativePerUnitLink
+    });
+    emit ConfigSet(
+      maxGasLimit,
+      stalenessSeconds,
+      gasAfterPaymentCalculation,
+      fallbackNativePerUnitLink,
+      gasOverhead,
+      donFee
+    );
+  }
+
+  /**
+   * @inheritdoc IFunctionsBilling
+   */
+  function getConfig()
+    external
+    view
+    returns (
+      uint32 maxGasLimit,
+      uint32 stalenessSeconds,
+      uint256 gasAfterPaymentCalculation,
+      int256 fallbackNativePerUnitLink,
+      uint32 gasOverhead,
+      address linkPriceFeed
+    )
+  {
+    return (
+      s_config.maxGasLimit,
+      s_config.stalenessSeconds,
+      s_config.gasAfterPaymentCalculation,
+      s_config.fallbackNativePerUnitLink,
+      s_config.gasOverhead,
+      address(LINK_TO_NATIVE_FEED)
+    );
+  }
+
+  // ================================================================
+  // |                 Cost Calculation Methods                     |
+  // ================================================================
+  /**
+   * @inheritdoc IFunctionsBilling
+   */
+  function getDONFee(
+    bytes memory /* data */,
+    RequestBilling memory /* billing */
+  ) public view override returns (uint96) {
+    // NOTE: Optionally, compute additional fee here
+    return s_config.donFee;
+  }
+
+  /**
+   * @inheritdoc IFunctionsBilling
+   */
+  function getAdminFee(
+    bytes memory /* data */,
+    RequestBilling memory /* billing */
+  ) public view override returns (uint96) {
+    // NOTE: Optionally, compute additional fee here
+    return IFunctionsRouter(address(s_router)).getAdminFee();
+  }
+
+  function getFeedData() public view returns (int256) {
+    uint32 stalenessSeconds = s_config.stalenessSeconds;
+    bool staleFallback = stalenessSeconds > 0;
+    (, int256 weiPerUnitLink, , uint256 timestamp, ) = LINK_TO_NATIVE_FEED.latestRoundData();
+    // solhint-disable-next-line not-rely-on-time
+    if (staleFallback && stalenessSeconds < block.timestamp - timestamp) {
+      weiPerUnitLink = s_config.fallbackNativePerUnitLink;
+    }
+    return weiPerUnitLink;
+  }
+
+  // ================================================================
+  // |                  Cost Estimation Methods                     |
+  // ================================================================
+  /**
+   * @inheritdoc IFunctionsBilling
+   */
+  function estimateCost(
+    uint64 subscriptionId,
```

Comment:
We shouldn't need the subId for estimateCost. Can we just fill in a dummy subId of 0 to create the RequestBilling struct?

Reply:
There is a world where we use it for user based pricing tiers, but we can just re-add it if that ever happens :man_shrugging:

Reply:
I'm more hesitant to remove this if we lock `estimateCost` into the Router interface.
Since it currently _is not_ in there I'm okay removing this.

Reply:
I'm going to keep as is for now because there is an incoming Economics change that may use this. I'd rather keep it as close to a real request as possible.

Reply:
👍